benchmarks/performance_test.py

```diff
@@ -1,17 +1,78 @@
 from urlparse4 import urlsplit, urljoin
 from timeit import default_timer as timer
 
-total = 0
```

Comment:
Since this file is changes so heavily in this PR, maybe also move all code into some function, e.g. ``main``, and call it at the end? There are two reasons:
- with a function, it's more natural to add helper functions when the code grows, and we avoid accidentally using global variables
- PyPy has much better optimizations for local function variables than for globals, so it would be more fair to measure performance inside a function

Reply:
It makes sense! I will move all the code inside a main() func